src/Akka.Persistence.Azure/Journal/AzureTableStorageJournal.cs

```diff
@@ -417,23 +378,20 @@ protected override void PreStart()
 
                     if (taggedEntries.Count > 0)
                     {
-                        var eventTagsBatch = new TableBatchOperation();
-
                         foreach (var kvp in taggedEntries)
                         {
-                            eventTagsBatch.Clear();
+                            var eventTagsBatch = new List<TableTransactionAction>();
 
                             foreach (var item in kvp.Value)
                             {
-                                item.PartitionKey = PartitionKeyEscapeHelper.Escape(item.PartitionKey);
-                                eventTagsBatch.InsertOrReplace(item);
+                                eventTagsBatch.Add(new TableTransactionAction(TableTransactionActionType.UpsertReplace, item.WriteEntity()));
                             }
 
-                            var eventTagsResults = await Table.ExecuteBatchAsLimitedBatches(eventTagsBatch);
+                            var eventTagsResponse = await Table.SubmitTransactionAsync(eventTagsBatch);
 
                             if (_log.IsDebugEnabled && _settings.VerboseLogging)
-                                foreach (var r in eventTagsResults)
-                                    _log.Debug("Azure table storage wrote entity [{0}] with status code [{1}]", r.Etag, r.HttpStatusCode);
+                                foreach (var r in eventTagsResponse.Value)
+                                    _log.Debug("Azure table storage wrote entity with status code [{1}]", r.Status);
 
                             if (HasTagSubscribers && taggedEntries.Count != 0)
```

Comment:
Should we get rid of the tag subscribers stuff inside the write loop? Seems out of place given that we still have to poll for changes on the table to detect changes that happen exclusively on other nodes in the network. This is something that got copied and pasted into a lot of journals but it was only ever really appropriate for SQLite / in-memory.

Reply:
We should, should we open another issue and PR for this?

Reply:
Yes we should